portfolio/src/main/webapp/script.js

```diff
@@ -197,27 +196,39 @@ function showCommentFormAndLoginPrompt() {
         const isLoggedIn = loginStatus.isLoggedIn;
         showCommentForm(isLoggedIn);
         showLoginOrLogoutForm(isLoggedIn);
+        showUserInfoFormAndGreeting(isLoggedIn);
       });
 }
 
 function showCommentForm(isLoggedIn) {
   const commentForm = document.getElementById('comment-form');
   if (isLoggedIn) {
-    commentForm.classList.remove('hidden');
+    // Show comment form after image URL is fetched.
+    fetch('/blobstore-upload-url')
+        .then((response) => {
+          return response.text();
+        })
+        .then((imageUploadUrl) => {
+          const commentForm = document.getElementById('comment-form');
+          commentForm.action = imageUploadUrl;
+          commentForm.classList.remove('hidden');
+        });
   } else {
     commentForm.classList.add('hidden');
   }
 }
 
-function showLoginOrLogoutForm(isLoggedIn) {
+function showLoginOrLogoutForm(isLoggedIn, email) {
```

Comment:
The new `email` parameter seems extraneous, since it's not used in the function and not an argument when it's called above

Reply:
Done - removed extraneous parameter email.